package.nix

```diff
@@ -1,49 +1,44 @@
 {
   lib,
+  hydra-check,
   rustPlatform,
-  pkg-config,
-  openssl,
-  stdenv,
-  installShellFiles,
+  versionCheckHook,
 }:
 
-rustPlatform.buildRustPackage {
-  pname = "hydra-check";
-  version = "2.0.0";
+hydra-check.overrideAttrs (
+  {
+    meta ? { },
+    nativeInstallCheckInputs ? [ ],
+    ...
+  }:
+  {
+    version = (builtins.fromTOML (builtins.readFile ./Cargo.toml)).package.version;
 
-  src = builtins.path {
-    name = "hydra-check-source";
-    path = ./.;
-  };
-
-  cargoLock = {
-    lockFile = builtins.path {
-      name = "hydra-check-Cargo.lock";
-      path = ./Cargo.lock;
+    # `builtins.path` works well with lazy trees
+    src = builtins.path {
+      name = "hydra-check-source";
+      path = ./.;
     };
-  };
 
-  nativeBuildInputs = [
-    pkg-config
-    installShellFiles
-  ];
+    cargoDeps = rustPlatform.importCargoLock {
+      lockFile = builtins.path {
+        name = "hydra-check-Cargo.lock";
+        path = ./Cargo.lock;
+      };
+    };
 
-  buildInputs = [
-    openssl
-  ];
+    nativeInstallCheckInputs = nativeInstallCheckInputs ++ [
+      versionCheckHook
+    ];
```

Comment:
This will be removed I guess once https://github.com/NixOS/nixpkgs/pull/378570 is merged?

Reply:
Yes! Once that is merged, I will bump the `flake.lock` here for nixpkgs and remove the redundant version check!